### DISCUSSION TITLE
@keyup.enter fires twice when combined with @click ?

### DISCUSSION THREAD

**Discussion by onemoreahmad** [alpinejs/alpine#4110] [7. Bugs (archived)] [answered]
I don't know if this is an expected behaviour or it's a bug, when I combined `@click` and `@keyup.enter` on a button, both functions run when I focus and hit Enter, the one inside keyup event AND the one in the click event. 

`<button x-data @click="console.log('clicked')" @keyup.enter="console.log('keyup enter')"> Click me </button>` 

now, when I click enter, I got on console "clicked", but when I focus on the button and hit Enter, I got both console clicked and keyup enter. I think it should run only the later. 

I use this to add new input item to a form array, when I click the button, one input added, but when I Tab and Focus on the add button and hit Enter, two inputs added! 

[Codepen link](https://codepen.io/onemoreahmad/pen/dyLzPGZ?editors=1111) 

<img width="869" alt="Screenshot 2024-03-26 at 8 45 30 PM" src="https://github.com/alpinejs/alpine/assets/1885785/b2887e3c-3167-456c-a914-4c1fb9bc0c17">

**Comment by onemoreahmad** [comment]
Hmm I think this is an expected behavior since browser by default fires click event when hit Enter on a button.

**Reply by SimoTod** [reply, marked as answer]
It's a standard behaviour. Because a button is not a text area, pressing enter while it's focused triggers a click (for accessibility reasons, everything should be possible by only using the keyboard).
Along with that, something that you may find even weirder, if your input is a child of a form element, and you press enter on a text input, that also triggers a click event on your button (and a consequent form submission). see https://codepen.io/SimoTod/pen/PogKZod?editors=1111 (I deliberately did not use Alpine for the example to show it's a browser behaviour)

**Reply by onemoreahmad** [reply]
Aha! interesting, thanks for the extra explanation. what I was trying to do is to `focus()` on the the new input that just created, but yes after removing that "@keyup.enter" and just put the focus action on the "@click" event, it did the trick (even with some issues related to Alpinejs!). Thanks again!

**Reply by ekwoka** [reply]
Similarly, using arrows to switch between radio inputs also dispatches click events from those radios.

Generally these click events will have no pointer type. like "touch" or "mouse" or "stylus". Just an empty string (but not always! damn safari)
